### PR TITLE
Bunch of changes to integrate cardbox auth in accesso

### PR DESCRIPTION
**Pull request by Drevoed**

### CODE REVIEW

File: api-internal/openapi.yaml [hunk 1]
```diff
@@ -138,6 +138,27 @@ paths:
         500:
           description: Something went wrong
 
+  "/client.get":
```
Comment (sergeysova) [inline api-internal/openapi.yaml:141]: Why you need an empty clientGet?
Why it returns SessionDeleteFailure?

File: api-public/src/generated.rs [hunk 2]
```diff
@@ -93,7 +93,7 @@ pub mod components {
 
             #[inline]
             fn from_request(req: &HttpRequest, _: &mut actix_web::dev::Payload) -> Self::Future {
-                match extract_header(req, "Authorization".to_string()) {
+                match extract_header(req, "X-Access-Token".to_string()) {
```
Comment (sergeysova) [inline api-public/src/generated.rs:96]: Better to change it back, and send a correct request from cardbox

File: core/src/services/email.rs [hunk 3]
```diff
@@ -35,6 +37,11 @@ impl From<SendGrid> for Email {
 impl EmailNotification for Email {
     #[tracing::instrument]
     async fn send(&self, email: String, message: EmailMessage) -> Result<(), SendEmailError> {
+        if !self.enabled {
+            tracing::warn!("Email service is disabled!");
```
Comment (sergeysova) [inline core/src/services/email.rs:41]: Maybe it is better to print message content to stdout?

**Review by sergeysova — CHANGES_REQUESTED** [review]

File: core/src/services/email.rs (hunk 3 above)
Comment (Drevoed) [reply core/src/services/email.rs:41]: Alongside warning with a level of DEBUG - yeah, why not.

File: api-internal/openapi.yaml [hunk 4]
```diff
@@ -138,6 +138,45 @@ paths:
...
+                  description: Application id
+      responses:
+        200:
+          description: Found application by id
+          content:
+            application/json:
+              schema:
+                required: [application]
+                properties:
+                  client:
```
Comment (sergeysova) [inline api-internal/openapi.yaml:164]:
```suggestion
                  application:
```

File: api-internal/openapi.yaml [hunk 5]
```diff
@@ -138,6 +138,45 @@ paths:
...
+  "/application.get":
+    post:
+      operationId: applicationGet
+      tags: [Application]
+      description: Get application info from id
+      requestBody:
+        content:
+          application/json:
+            schema:
+              required: [ clientId ]
```
Comment (sergeysova) [inline api-internal/openapi.yaml:150]:
```suggestion
              required: [ applicationId ]
```

File: api-internal/src/routes/application/get.rs [hunk 6]
```diff
@@ -0,0 +1,44 @@
+use crate::generated::{
+    components::{
+        request_bodies::ApplicationGetRequestBody,
+        responses::{
+            ApplicationGetError as FailureVariant, ApplicationGetFailure as Failure,
+            ApplicationGetSuccess,
+        },
+        schemas::Client,
```
Comment (sergeysova) [inline api-internal/src/routes/application/get.rs:8]:
```suggestion
        schemas::Application,
```

**Review by sergeysova — CHANGES_REQUESTED** [review]

**Review by sergeysova — APPROVED** [review]

**Pull request merged**